swap_meet/vendor.py

```diff
@@ -1,2 +1,75 @@
+from swap_meet.item import Item
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self,inventory=None):
+        if not inventory:
+            inventory = []
+        self.inventory = inventory
+    
+    def add(self,new_item):
+        self.inventory.append(new_item)
+        return new_item
+
+    def remove(self,new_item):
+        try:
+            self.inventory.remove(new_item)
+            return new_item
+        except ValueError as error:
+            print(f"{new_item} not in inventory")
+            return False
+
+    def get_by_category(self,category):
+        category_list = []
+        try:
+            for item in range(len(self.inventory)):
+                if self.inventory[item].category is category:
+                    category_list.append(self.inventory[item])
+        except ValueError as error:
+            print(f"There are no {category} in inventory")
+        return category_list
+
+    def swap_items(self,Friend,my_item,their_item):
+        index1 = 0
+        index2 = 0
+        if my_item not in self.inventory or their_item not in Friend.inventory:
+            return False
+        for x in range(len(self.inventory)):
+            if self.inventory[x] is my_item:
+                index1 = x
+        for y in range(len(Friend.inventory)):
+            if Friend.inventory[y] is their_item:
+                index2 = y
```

Comment:
look into the python built-in method `index()`https://www.w3schools.com/python/ref_list_index.asp